logstash-core/lib/logstash/instrument/metric_type/counter.rb

```diff
@@ -1,29 +1,17 @@
-# encoding: utf-8
-require "logstash/instrument/metric_type/base"
-require "concurrent"
+#encoding: utf-8
+java_import org.logstash.instrument.metrics.counter.LazyDelegatingCounter
```

Comment:
OK, I will remove the DoubleCounter and restrict the Java code to a Long type and ensure that the lazy instantiation will throw an exception if not Long type.  I too was having trouble thinking of a valid use case for it.